protocol/messenger.go

```diff
@@ -2527,15 +2527,14 @@ func (m *Messenger) SyncDevices(ctx context.Context, ensName, photoPath string,
 	}
 
 	m.allChats.Range(func(chatID string, chat *Chat) (shouldContinue bool) {
-		isPublicChat := !chat.Timeline() && !chat.ProfileUpdates() && chat.Public()
-		if isPublicChat && chat.Active {
-			err = m.syncPublicChat(ctx, chat, rawMessageHandler)
+		if chat.shouldBeSynced() {
+			err = m.syncChat(ctx, chat, rawMessageHandler)
 			if err != nil {
 				return false
 			}
 		}
 
-		if (isPublicChat || chat.OneToOne() || chat.PrivateGroupChat()) && !chat.Active && chat.DeletedAtClockValue > 0 {
+		if (chat.Public() || chat.OneToOne() || chat.PrivateGroupChat()) && !chat.Active && chat.DeletedAtClockValue > 0 {
```

Comment:
I am pretty sure that we can remove entire this condition (lines 2537 - 2547), cause all that will be properly synced via `err = m.syncChat(ctx, chat, rawMessageHandler)` in the line `2531`.

The same states for the 2 blocks below, that can be removed as well:
- lines 2551 - 2556
- lines 2558 - 2563

Also in terms of local pairing these messages should arrive to the destination in the order they are sent, but this `SyncDevice` function may be used to broadcast sync data over the wire in that case no guarantee of the message order and in that case handling "messages read" or "clear history" related sync messages doesn't have point if we're not sure that the chat those sync messages are referring to is not received.

Reply:
I removed the block at 2537. I agree that it's useless now that we add `Active` in the Chat itself.

I get your point about the order of message for "message read" and "clear history", but in case it does arrive in order, isn't it better to have it than not?

Reply:
> in case it does arrive in order, isn't it better to have it than not?

I would say that we don't need them at all, but also keeping them won't ruin anything. The reason why I think so is cause nothing has changed in the meantime. 

If we're setting up device B (that has an empty state at this moment) from device A (that has some data, like chats, contacts, accounts, whatever...) when local pairing (syncing) starts there is no way anything to be changed by user's activity that can change the context of the data being synced. Anything changed after the syncing process will be synced, cause at that point in time devices A and B are paired. That means we only need to sync the state, not the events. Sync messages sent via `syncChatMessagesRead` and `syncClearHistory` are initiated by the user's activity.

That was my point, why keep the code, that won't have any effect on the other side, just send a few messages more.

Reply:
Ok yeah I get your point. It makes sense.